OneCommunityGlobal/HGNRest
# Aishwarya_Purchase-Tool-Request-Form-Backend

Aish_Purchase-Tool-Request-Form-(Backend)

Description
This pull request adds the backend logic for the Purchase Request: Tool form in the BM Dashboard (Phase 2).

Related PRS (if any):
This PR must be checked out to test #1752 frontend PR

Main changes explained:
Add BM Dashboard collections to the dev database: buildingTool
Add Mongoose schema file buildingTool
Add POST route to either create a new document in buildingTool collection (if the tool is being ordered to the project for the first time), or update purchase record sub-document in an existing document (if the material has been ordered previously for the project). In either case, the purchaseRecord sub-document will have default values added to date, requestedBy, and status fields, in addition to the submitted form data.

How to test:
git checkout Aish_Purchase-Tool-Request-Form-(Backend)
npm install
npm run dev
check the related frontend PR for further instructions

Optional: connect to the dev db via Mongo Compass, open the buildingTool collection, and observe that documents and sub-documents are being updated correctly.

Notes
Please remember to either Approve or Request Changes. Do not just leave a comment.
Please leave a review for both the front and backend branches after you have finished testing!
